api/src/main/java/com/datatorrent/api/DAG.java

```diff
@@ -288,4 +288,28 @@
   {
 
   }
+
+  /**
+   * Implementation of {@see DAG} is expected to call all visitors registered
+   * after whole DAG is populated by the application. Implementation is free to
+   * perform other transformations before calling this API. Each visitor should
+   * get called in the order they are registered one after the other. i.e only
+   * after one visitor has examine the DAG, the next visitor can start examining
+   * the DAG. The order in which individual operators and streams are visited are
+   * left to the implementation of DAG, i.e visitor should not make any assumptions
+   * about the order in which {@see DAG.Visitor.visitOperator}, {@see DAG.Visitor.visitStream}
+   * is called.
+   */
+  @InterfaceStability.Evolving
+  interface Visitor extends Serializable
+  {
+    void preVisitDAG(DAG dag);
+
+    void visitOperator(OperatorMeta ometa);
```

Comment:
@sandeshh  it is possible to access operator properties as OperatorMeta can give you reference to the instance of the operator.

@PramodSSImmaneni  I have updated the pull request after moving this interface to separate file.